src/main/java/net/shortninja/staffplus/core/common/cmd/AbstractCmd.java

```diff
@@ -84,6 +85,10 @@ public boolean execute(CommandSender sender, String alias, String[] args) {
             String playerName = getPlayerName(sender, filteredArgs).orElse(null);
 
             Optional<SppPlayer> player = commandService.retrievePlayer(sppArgs, playerName, getPlayerRetrievalStrategy(), delayable);
+            
+            if (player == null) {

```

Comment:
I don't think this check is in the correct place. If a playername is mandatory the command itself should specify it as such playerRetrievalStrategy = ONLINE or playerRetrievalStrategy = BOTH
if the command specifies it and the user still provides a null value the CommandPlayerRetriever class could throw and exception directly in that case rather than returning null

Reply:
Changed it to a `IllegalArgumentException`.